linkcheck/plugins/parsepdf.py

```diff
@@ -44,14 +44,7 @@ def search_url(obj, url_data, pageno, seen_objs):
     if isinstance(obj, dict):
         for key, value in obj.items():
             if key == 'URI':
-                if isinstance(value, str):
-                    url = value
-                else:
-                    # URIs should be 7bit ASCII encoded, but be safe and encode
-                    # to unicode
-                    # XXX this does not use an optional specified base URL
-                    url = strformat.unicode_safe(value)
-                url_data.add_url(url, page=pageno)
+                url_data.add_url(str(value, encoding="ascii"), page=pageno)
```

Comment:
Whoa, I don't think I've ever seen `str(something, encoding='ascii')` before.  `something.decode('ascii')` is the canonical spelling of this operation.

Reply:
Done.